File: bindings/ruby/samples/tbit_test.rb
#!/usr/bin/env ruby

$:.unshift File.join(File.dirname(__FILE__), ".")
require 'tbi'

tbit = Tbi::Tbit.new

tbit.gpio.digitalWrite(15, 1);
puts sprintf("%.32b ", tbit.gpio.read())
sleep(0.5)
tbit.gpio.digitalWrite(15, 0);
puts sprintf("%.32b ", tbit.gpio.read())

puts sprintf("%.32b ", tbit.adc.analogRead(0))

led = Tbi::Pin.new(tbit.gpio, 15)
outpin = Tbi::Pin.new(tbit.gpio, 6, Tbi::OUTPUT_PIN)
inpin = Tbi::Pin.new(tbit.gpio, 5, Tbi::INPUT_PULLUP_PIN)

sleep(0.5)
led.on()
outpin.on()
puts inpin.value()

sleep(0.5)
led.off()
outpin.off()
puts inpin.value()

File: bindings/python/samples/tbit_test.py
import tbi
import time

tbit = tbi.Tbit()

print tbit.gpiohw.read()
tbit.gpiohw.digitalWrite(15, True)
time.sleep(0.5)
tbit.gpiohw.digitalWrite(15, False)

led = tbi.Pin(tbit, 15, tbi.OUTPUT_PIN)
outpin = tbi.Pin(tbit, 6, tbi.OUTPUT_PIN)
inpin = tbi.Pin(tbit, 5, tbi.INPUT_PULLUP_PIN)
adcpin = tbi.Adc(tbit, 4)

time.sleep(0.5)
led.on()
outpin.on()
print inpin.value()
print adcpin.value()

time.sleep(0.5)
led.off()
outpin.off()
print inpin.value()
print adcpin.value()

